### ISSUE TITLE
It would be nice if Checkbox instances could be disabled (while still visible)

### ISSUE THREAD

**Issue by imbuedhope** [Textualize/textual#1679]
I'm working a small app with a several checkboxes that can be toggled before starting a long running process. When the process is running it doesn't make sense for the checkboxes to be toggleable.

I subclassed `Checkbox` and got it working, but I think it makes sense for this to be a default feature.

My quick / lazy implementation was as follows
```python3
class LockableCheckbox(Checkbox):
    def __init__(self, value: bool = False, *, enabled: bool = True,
        animate: bool = True, name: str | None = None, id: str | None = None,
        classes: str | None = None):

        super().__init__(
            value=value, animate=animate, name=name, id=id, classes=classes
        )
        self.enabled = enabled

    def on_click(self) -> None:
        if self.enabled:
            self.toggle()
```

**Comment by davep** [comment]
This should end up being resolved once I get #1748 under way.

As an aside to the request here, heads-up that what is currently called a `Checkbox` will be renamed to `Switch` in the next release. Same interface and working, just a different name to make way for some other work.

**Comment by davep** [comment]
@imbuedhope Just to update: all widgets in Textual now have a `disabled` property, and setting it to `True` disables that widget and any children it may have too. On top of that we've also revamped the `Checkbox` (moving the old version to `Switch`) and have also added `RadioButton` and `RadioSet`.

Thanks again for the issue; hope this helps.
